### PR TITLE
Bugfix/children change does not rerender

### LINKED ISSUE #424: Broken Button With React after master merge
`<AXAButtonReact onAxaClick={this.handleClick} key={1} >{this.state.isToggleOn ? 'ON' : 'OFF'}</AXAButtonReact>` does not update (always ON is shown. Is ONLY appeared after master merge as it was tested before mergin and worked correctly

on localhost

### PR DESCRIPTION
**Pull request by AndyOGo**
Fixes #424 .

Changes proposed in this pull request:

 - sync updated children (light DOM) of react integration
 -
 -

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AndyOGo** [comment]
> Uncaught TypeError: Cannot read property 'replaceChild' of null

**Comment by LucaMele** [comment]
when typing focus is lost

bildschirmfoto 2018-04-12 um 16 56 45

File: src/js/with-react.jsx [hunk 1]
```diff
@@ -53,16 +55,28 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
...
     }
 
-    componentWillReceiveProps(props) {
+    componentDidUpdate() {
+      const { props, slot, wcNode } = this;
+      const { children } = props;
+      const slotClone = slot.cloneNode(false);
+
+      // @todo: this rebuilds the children's DOM tree every time, NO DIFFING
+      ReactDOM.render(children, slotClone, () => {
```
Comment (AndyOGo) [inline src/js/with-react.jsx]: @LucaMele 
This line disables React'S diffing by always creating new children's DOM tree in a temporary `<div>`

**Comment by AndyOGo** [comment]
yeah that's not an option

**Comment by AndyOGo** [comment]
@LucaMele 
I disabled `isSameNodeOnce` now and suddenly it works.
May I have some miss-conception about some details.

But for now just disabling it, if we are sure what really happens underhood we can remove it or find a better solution.

File: src/js/abstract/is-same-node-once.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ let sameNodeCache = [];
  * @param node
  */
 export function isSameNodeOnce(node) {
```
Comment (LucaMele) [inline src/js/abstract/is-same-node-once.js]: lets delete this and make an issue so we dont forget and also why

**Pull request merged**